src/java/org/apache/cassandra/service/snapshot/SnapshotManifest.java

```diff
@@ -60,13 +60,18 @@ private SnapshotManifest() {
         this.expiresAt = null;
     }
 
-    public SnapshotManifest(List<String> files, Duration ttl)
+    public SnapshotManifest(List<String> files, Duration ttl, long timestamp)
     {
         this.files = files;
-        this.createdAt = Instant.now();
+        this.createdAt = Instant.ofEpochMilli(timestamp);
         this.expiresAt = ttl == null ? null : createdAt.plusMillis(ttl.toMilliseconds());
     }
 
+    public SnapshotManifest(List<String> files, Duration ttl)
```

Comment:
Here is a conceptual question - do we still need to have this constructor? It is only used in tests. We can update tests to use the new one. If we remove this constructor we will enforce developers to always pass a timestamp, in turn that will not let them introduce the same issue as you're fixing now.

The same is applicable to a few other methods you've updated.

Reply:
thanks for review @alex-ninja ,